adoc/chapters/programming_interface.adoc

```diff
@@ -963,207 +963,237 @@ are handled by throwing synchronous SYCL exceptions.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, additional exceptions that are thrown, etc.
+
+'''
+
+.[apidef]#platform::has#
+[source,role=synopsis,id=api:platform-has]
 ----
 bool has(aspect asp) const
 ----
-   a@ Returns true if all of the SYCL [code]#devices# associated with this
-      SYCL [code]#platform# have the given <<aspect>>. Returns [code]#false# if this SYCL [code]#platform# does not
-      contain any devices.
 
-a@
-[source]
+_Returns:_ The value [code]#true# if all of the devices associated with this
+platform have the given <<aspect>>.
+Returns the value [code]#false# if this platform does not contain any devices.
+
+'''
+
+.[apidef]#platform::has_extension#
+[source,role=synopsis,id=api:platform-has-extension]
 ----
 bool has_extension(const std::string& extension) const
 ----
-   a@ Deprecated, use [code]#has()# instead.
 
-Returns [code]#true# if this SYCL [code]#platform# supports the extension queried by the [code]#extension# parameter. A SYCL [code]#platform# can only support an extension if all associated SYCL [code]#devices# support that extension. Returns [code]#false# if this SYCL [code]#platform# does not contain any devices.
+Deprecated by SYCL 2020.
 
-a@
-[source]
+{note}Use [api]#platform::has# instead.
+{endnote}
+
+_Returns:_ The value [code]#true# if this platform supports the extension
+queried by the [code]#extension# parameter.
+A platform only supports an extension if all associated devices support that
+extension.
+Returns [code]#false# if this platform does not contain any devices.
+
+'''
+
+.[apidef]#platform::get_devices#
+[source,role=synopsis,id=api:platform-get-devices]
 ----
 std::vector<device>
-get_devices(info::device_type deviceType = info::device_type::all) const
+get_devices(info::device_type type = info::device_type::all) const
 ----
-   a@ Returns a [code]#std::vector# containing all the
-      <<root-device, root devices>> associated with this SYCL [code]#platform#
-      which have the device type encapsulated by [code]#deviceType#.
 
-|====
+_Returns:_ A [code]#std::vector# containing all of the <<root-device,root
+devices>> associated with this platform which have the device type specified by
+[code]#type#.
 
+'''
 
+[[sec:platform-static-member-funcs]]
+==== Static member functions
 
-[[table.staticmembers.platform]]
-.Static member functions of the SYCL [code]#platform# class
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Static member function @ Description
-a@
-[source]
+.[apidef]#platform::get_platforms#
+[source,role=synopsis,id=api:platform-get-platforms]
 ----
 static std::vector<platform> get_platforms()
 ----
-   a@ Returns a [code]#std::vector# containing all SYCL
-      [code]#platforms# from all <<backend, SYCL backends>> available in the system.
-
-|====
 
+_Returns:_ A [code]#std::vector# containing all of the platforms from all
+<<backend, backends>> that are known to the SYCL implementation.
```

Comment:
I'm not sure "known" and "available" mean the same thing here. Would we expect this API to return backends that are currently useable at the time this is called. That is different to an implementation having a backend for devices that are not present in the system. For example, an implementation might provide a CUDA backend, but if the system has no NVIDIA GPUs, I don't think implementations show it.

Reply:
I restored the wording about "available" platforms in c1780c506780b44f6b85cf3a83d2b589ef2a603d.

However, I wonder if this is really what we want.  The change we made in #468 (issue #459) was done specifically so that `get_platforms` could potentially return an "empty" platform for the case when the implementation supports a backend, but there are no available devices in that backend at runtime.

Reply:
I think "available" covers that case. Thanks for the change.